lib/loggly/common.js

```diff
@@ -209,20 +226,28 @@ common.loggly = function () {
     }
     try {
       request(requestOptions, function (err, res, body) {
-        if (err) return onError(err);
-        var statusCode = res.statusCode.toString();
-        if(statusCode === '403') isValidToken = false;
-        if (statusCode === '500' || statusCode === '503' || statusCode === '504') retryOnServerError(res);
-        if (statusCode === '200') {
-          arrRetryLogs.splice(0, arrSize);
-          totalRetries = 0;
-        }
-        if (Object.keys(failCodes).indexOf(statusCode) !== -1) {
-          if (statusCode !== '503' && statusCode !== '500' && statusCode !== '504') {
-            return onError((new Error('Loggly Error (' + statusCode + '): ' + failCodes[statusCode])));
+        if (err) {
+          // In rare cases server is busy
+          if (err.code === 'ETIMEDOUT' || err.code === 'ECONNRESET' || err.code === 'ESOCKETTIMEDOUT' || err.code === 'ECONNABORTED') {
+            retryOnError(bulk, err);
+          } else {
+            return onError(err);
+          }
+        } else {
+            responseCode = res.statusCode;
+            if (responseCode === httpStatusCode.badToken.code) {
+              isValidToken = false;
+              return onError((new Error('Loggly Error (' + responseCode + '): ' + httpStatusCode.badToken.message)));
+            }
+            if (responseCode === httpStatusCode.success.code && bulk.attemptNumber >= eventRetried) {
```

Comment:
@mchaudhary This condition checks if the event sent successfully after retrying "n" number of times with status code 200 then print a console message.